event-sourcing/event-source-api/src/main/java/uk/gov/justice/services/eventsourcing/source/api/service/EventsPageService.java

```diff
@@ -0,0 +1,106 @@
+package uk.gov.justice.services.eventsourcing.source.api.service;
+
+import static uk.gov.justice.services.eventsourcing.repository.jdbc.Direction.BACKWARD;
+import static uk.gov.justice.services.eventsourcing.repository.jdbc.Direction.FORWARD;
+import static uk.gov.justice.services.eventsourcing.repository.jdbc.Position.first;
+import static uk.gov.justice.services.eventsourcing.repository.jdbc.Position.head;
+import static uk.gov.justice.services.eventsourcing.repository.jdbc.Position.position;
+
+import uk.gov.justice.services.eventsourcing.repository.jdbc.Direction;
+import uk.gov.justice.services.eventsourcing.repository.jdbc.FixedPosition;
+import uk.gov.justice.services.eventsourcing.repository.jdbc.Position;
+import uk.gov.justice.services.eventsourcing.source.api.feed.common.Page;
+import uk.gov.justice.services.eventsourcing.source.api.feed.common.PagingLinks;
+import uk.gov.justice.services.eventsourcing.source.api.feed.event.EventEntry;
+
+import java.net.MalformedURLException;
+import java.net.URL;
+import java.sql.SQLException;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.UUID;
+
+import javax.enterprise.context.ApplicationScoped;
+import javax.inject.Inject;
+import javax.ws.rs.core.UriInfo;
+
+/**
+ * Generator of paginated events
+ */
+@ApplicationScoped
+public class EventsPageService {
+
+    private static final int PATH_SEGMENT_EVENT_STREAMS = 0;
+
+    private static final int PATH_SEGMENT_EVENT_STREAM_ID = 1;
+
+    private static final long ZERO = 0L;
+
+    @Inject
+    private EventsService service;
+
+    public Page<EventEntry> pageEvents(final String position,
+                                       final Direction direction,
+                                       final long pageSize,
+                                       final UriInfo uriInfo,
+                                       final UUID streamId) throws SQLException, MalformedURLException {
+
+        final Position positionOf = position(position);
+
+        final List<EventEntry> entities = service.events(positionOf, direction, pageSize, streamId);
```

Comment:
i'd change the signature of the service, have the streamID, position, direction, pageSize.

Again below (feedWithLinks), it makes more sense to have the key information at the front.

Reply:
Changed